pkg/operator/operator.go

```diff
@@ -169,13 +179,29 @@ func NewOperator(ctx context.Context, operator *operator.Operator) (context.Cont
 		SecurityGroupProvider:     securityGroupProvider,
 		AMIProvider:               amiProvider,
 		AMIResolver:               amiResolver,
+		VersionProvider:           versionProvider,
 		LaunchTemplateProvider:    launchTemplateProvider,
 		PricingProvider:           pricingProvider,
 		InstanceTypesProvider:     instanceTypeProvider,
 		InstanceProvider:          instanceProvider,
 	}
 }
 
+func (op *Operator) ValidateK8sVersion(ctx context.Context) error {
+	v := lo.Must1(op.VersionProvider.Get(ctx))
+	k8sVersion := versionv1.MustParseGeneric(v)
+
+	// We will only error if the user is running karpenter on a k8s version,
+	// that is out of the range of the minK8sVersion and maxK8sVersion
+	minCompare := lo.Must1(k8sVersion.Compare(minK8sVersion))
+	maxCompare := lo.Must1(k8sVersion.Compare(maxK8sVersion))
+	if minCompare == 1 && maxCompare == -1 {
```

Comment:
nit: There's `k8sVersion.LessThan` and `k8sVersion.AtLeast` to make this more idiomatic.